visualization/dashboard/src/MLIDashboard/Controls/FilterControl.tsx

```diff
@@ -19,87 +21,118 @@ export interface IFilterControlState {
-                    onClick={this.openFilter.bind(this, index)}>{this.props.jointDataset.metaDict[filter.column].abbridgedLabel}
-                </div>
+            return (<div key={index} className={FilterControl.classNames.existingFilter}>
+                    <Text variant={"small"} className={FilterControl.classNames.filterLabel}>{this.props.jointDataset.metaDict[filter.column].abbridgedLabel 
+                    + " " + filter.method + " " + filter.arg}</Text>
                 <IconButton
+                    className={FilterControl.classNames.filterIcon}
+                    iconProps={{iconName:"Edit"}}
+                    onClick={this.editFilter.bind(this, filter, index)}
+                />
+                <IconButton
+                    className={FilterControl.classNames.filterIcon}
                     iconProps={{iconName:"Clear"}}
                     onClick={this.removeFilter.bind(this, index)}
                 />
             </div>);
         });
-        return(
-            <div className={FilterControl.classNames.filterList}>
-                <Button
-                    onClick={this.editNewFilter}
-                    text="Add Filter"
-                />
-                {this.state.openedFilter !== undefined && (<FilterEditor
-                    jointDataset={this.props.jointDataset}
-                    onAccept={this.updateFilter}
-                    onCancel={this.cancelFilter}
-                    initialFilter={this.state.openedFilter}
-                />)}
-                {filterList}
+
+        return(<div className={FilterControl.classNames.wrapper}>
+            <FilterEditor
+                jointDataset={this.props.jointDataset}
+                onAccept={this.updateFilter}
+                onCancel={this.cancelFilter}
+                initialFilter={this.state.openedFilter}
+            />
+
+        <Text variant={"medium"} className={FilterControl.classNames.addedFilter} >Added Filters</Text>
+        <div className={FilterControl.classNames.addedFilterDiv}>
+        {filterList.length>0
+            ? <div>{filterList}</div>
+            : <div>
+                <Text variant={"smallPlus"} className={FilterControl.classNames.defaultFilterList}>No filters added yet</Text>
             </div>
-        );
+        }
+        </div>     
+        </div>
+);
     }
 
     private updateFilter(filter: IFilter): void {
         const index = this.state.filterIndex;
         this.setState({openedFilter: undefined, filterIndex: undefined});
         this.props.filterContext.onUpdate(filter, index);
+        this.forceUpdate();
     }
 
     private cancelFilter(): void {
         this.setState({openedFilter: undefined, filterIndex: undefined});
     }
 
-    private editNewFilter(): void {
-        this.setState({openedFilter: this.initialFilter, filterIndex: this.props.filterContext.filters.length});
+    private removeFilter(index: number): void {
+        this.props.filterContext.onDelete(index);
```

Comment:
The filter context has changed now that we use cohorts. Now, this modal is open with a cohort to be edited (ICohort with a name: string, filterList: IFilter[]) It should hold that all in state, and on add, edit, delete, change the filterList state object accordingly. Again, to have the component re-render, you'll do newFilterList = [...filterList].push(newFilter) or just slice for delete. That will create a new copy of the filter array, and trigger change